File: 백준/Gold/11657. 타임머신/타임머신.js
const fs = require("fs");

const reader = {
  data: fs
    .readFileSync(process.platform === "linux" ? 0 : "input.txt", "utf-8")
    .toString()
    .trim()
    .split("\n"),

  cursor: 0,

  read() {
    return this.data[this.cursor++];
  },

  readIntArray() {
    return this.read().split(" ").map(s => parseInt(s));
  }
}

function solve() {
  const [N, M] = reader.readIntArray();

  const edges = new Array(M).fill(null);

  for (let i = 0; i < M; i++) {
    edges[i] = reader.readIntArray();
  }

  const distance = new Array(N + 1).fill(Infinity);
  distance[1] = 0;

  for (let i = 0; i < N - 1; i++) {
    for (const [start, end, weight] of edges) {
      if (distance[end] > distance[start] + weight) {
        distance[end] = distance[start] + weight;
      }
    }
  }

  for (const [start, end, weight] of edges) {
    if (distance[end] > distance[start] + weight) {
      return -1;
    }
  }

  return distance
    .slice(2)
    .map(value => value === Infinity ? -1 : value)
    .join("\n");
}

console.log(solve());

File: 백준/Gold/11657. 타임머신/타임머신.py
import sys

input = sys.stdin.readline
print = sys.stdout.write

nr_cities, nr_routes = map(int, input().split())
edges = []
distance = [sys.maxsize] * (nr_cities + 1)

distance[1] = 0

for _ in range(nr_routes):
    edges.append(list(map(int, input().split())))

for _ in range(nr_cities - 1):
    for start, end, weight in edges:
        if distance[start] != sys.maxsize and distance[end] > distance[start] + weight:
            distance[end] = distance[start] + weight

cycle = False

for start, end, weight in edges:
    if distance[start] != sys.maxsize and distance[end] > distance[start] + weight:
        cycle = True
        break

if cycle:
    print("-1")
else:
    for i in range(2, nr_cities + 1):
        if distance[i] == sys.maxsize:
            print("-1\n")
        else:
            print(f"{distance[i]}\n")
